traversal/scanner/GraphIterator.java

```diff
@@ -176,82 +214,29 @@ private boolean computeAnswer() {
 
     private void traverse(ProcedureVertex<?, ?> procedureVertex) {
         VertexTraverser vertexTraverser = vertexTraversers.get(procedureVertex);
-        boolean verified = false;
-        Set<ProcedureVertex<?, ?>> verifyFailureCauses = new HashSet<>();
-        while (!verified && vertexTraverser.hasNextVertex()) {
-            vertexTraverser.takeNextVertex();
-            if (vertexTraverser.verifyLoops()
-                    && verifyTraversedScopes(procedureVertex, verifyFailureCauses)
-                    && verifyEdgeLookahead(procedureVertex, verifyFailureCauses)) {
-                verified = true;
-            } else {
-                vertexTraverser.clearCurrentVertex();
-                removeFromSuccessors(procedureVertex);
-            }
-        }
-        toRevisit.addAll(verifyFailureCauses);
-        if (verified) procedureVertex.outs().forEach(edge -> toTraverse.add(edge.to()));
-        else failed(procedureVertex);
+        if (vertexTraverser.findNextVertex()) success(vertexTraverser);
+        else failed(vertexTraverser);
```

Comment:
vertexTraversers now do a search internally, rather than having the `while` loop externally